pkg/gateway/types/oauthapps.go

```diff
@@ -176,10 +204,12 @@ func MergeOAuthAppManifests(r, other types.OAuthAppManifest) types.OAuthAppManif
 	if other.ClientSecret != "" {
 		retVal.ClientSecret = other.ClientSecret
 	}
-	if other.AuthURL != "" {
+	if other.AuthURL != "" && other.GitLabBaseURL == "" {
+		// Only keep custom auth URL if GitLabBaseURL isn't changing
 		retVal.AuthURL = other.AuthURL
 	}
-	if other.TokenURL != "" {
+	if other.TokenURL != "" && other.GitLabBaseURL == "" {
+		// Only keep custom token URL if GitLabBaseURL isn't changing
 		retVal.TokenURL = other.TokenURL
 	}
```

Comment:
is `other.GitLabBaseURL` necessary here? what edge case does this protect against?